### DISCUSSION TITLE
Error: cannot estimate gas when fulfill an offer

### DISCUSSION THREAD

**Discussion by yihsiulee** [ProjectOpenSea/seaport-js#322] [Q&A]
Hi, 

I am trying to build a function that can `Listing an ERC-721 and fulfilling it` and `Making an offer and fulfilling it`.
Env: polygon Mumbai testnet

It's successful when I list an ERC721 and fulfilling the order.

But I got an error when I called `seaport.fulfillOrder` to fulfill an offer.
error:

`Uncaught (in promise) Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted", method="estimateGas", transaction={"from":"0xeeEFbDFa58878e73ac087325a41EdFdf12eF4dbd","to":"0x00000000000000ADc04C56Bf30aC9d3c0aAF14dC","value":{"type":"BigNumber","hex":"0xe8d4a51000"},"data":"0xb3a34c4c0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000`

my code for making an offer and fulfilling the order

make an offer:

```typescript
const { executeAllActions } = await seaport.createOrder(
      {
        offer: [
          {
            amount: ethers.utils.parseEther(inputAmount).toString(),
          },
        ],
        consideration: [
          {
            itemType: ItemType.ERC721,
            token: inputAddress,
            identifier: inputId,
            recipient: address,
          },
        ],
      },
      address
    );
    const order = await executeAllActions();
```

fulfill the order:

```typescript
 const { executeAllActions: executeAllFulfillActions } =
      await seaport.fulfillOrder({
        order: order,
        accountAddress: fulfiller,
      });

    await executeAllFulfillActions();
  };
```

Does anyone have any idea where is the problem? Thanks.

**Comment by ryanio** [comment]
"cannot estimate gas" is a generic error that the tx will revert. i suggest adding a manual gasLimit of like 300_000 gas and seeing if it returns a proper error, which you can query the 4 byte code with https://openchain.xyz/signatures to see what it could be. tenderly can help as well to simulate to find the trace of the error.

your code sample is not complete enough to know what is going wrong, as you haven't defined `currentOrder`. what happens when you console.log it. since you are typecasting it with `as` it might not be the proper format.

**Reply by yihsiulee** [reply]
@ryanio sorry I didn't explain it well. The `currentOrder` is the `order` and I also edit the original post. 
And I remove the type `as ` `OrderWithCounter ` but still error.

I would like to know where I can set the gasLimit, due to even the metamask didn't pop out when I use the `fulfillOrder` function.
Also, the transaction didn't happen.

and
Here is the complete error code, and forgot to show you the error message `message":"Internal JSON-RPC error.",`
`Uncaught (in promise) Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted", method="estimateGas", transaction={"from":"0xeeEFbDFa58878e73ac087325a41EdFdf12eF4dbd","to":"0x00000000000000ADc04C56Bf30aC9d3c0aAF14dC","value":{"type":"BigNumber","hex":"0xe8d4a51000"},"data":"0xb3a34c4c00000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000004c00000000000000000000000001d75e36227d5455e6ad7b33addd7f0092142e6f100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000160000000000000000000000000000000000000000000000000000000000000022000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000064c25998ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002b1f35d6d41fd5ac000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000030000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000009184e72a000000000000000000000000000000000000000000000000000000009184e72a00000000000000000000000000000000000000000000000000000000000000000030000000000000000000000000000000000000000000000000000000000000002000000000000000000000000304cb9f97e604afedaf2655c1f784df9f97155e30000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000010000000000000000000000001d75e36227d5455e6ad7b33addd7f0092142e6f1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000746a528800000000000000000000000000000000000000000000000000000000746a5288000000000000000000000000003052a3c5402cb64f7c6eb69a4cfeadadae7ac2a0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000746a528800000000000000000000000000000000000000000000000000000000746a528800000000000000000000000000fef1cc5906e538d5eb956d92078050d4504583bf0000000000000000000000000000000000000000000000000000000000000040a98d4cdc84da09118393c63d1fa300134cf47344f1e2171dda79b945491b394bdaa9125c4732e5af12d34713c01817916ebd8420a66c42e03bb206b438b2a262","accessList":null}, error={"code":-32603,"message":"Internal JSON-RPC error.","data":{"code":3,"message":"execution reverted","data":"0xbc806b96000000000000000000000000eeefbdfa58878e73ac087325a41edfdf12ef4dbd000000000000000000000000000000000000000000000000000009184e72a000"},"stack":"{\n  \"code\": -32603,\n  \"message\": \"Internal JSON-RPC error.\",\n  \"data\": {\n    \"code\": 3,\n    \"message\": \"execution reverted\",\n    \"data\": \"0xbc806b96000000000000000000000000eeefbdfa58878e73ac087325a41edfdf12ef4dbd000000000000000000000000000000000000000000000000000009184e72a000\"\n  },\n  \"stack\": \"Error: Internal JSON-RPC error.\\n   
`
Thank you for your kind reply.
